unistd/dir.c

```diff
@@ -368,20 +368,25 @@ DIR *opendir(const char *dirname)
 	msg_t msg = { 0 };
 	char *canonical_name = resolve_path(dirname, NULL, 1, 0);
 	DIR *s = calloc(1, sizeof(DIR));
+	oid_t mnt;
 
 	if ((canonical_name == NULL) || (s == NULL)) {
 		free(canonical_name);
 		free(s);
 		return NULL; /* errno set by resolve_path */
 	}
 
-	if (!dirname[0] || (safe_lookup(canonical_name, &s->oid, NULL) < 0)) {
+	if (!dirname[0] || (safe_lookup(canonical_name, &s->oid, &mnt) < 0)) {
 		free(canonical_name);
 		free(s);
 		errno = ENOENT;
 		return NULL; /* ENOENT */
 	}
 
+	/* TODO: add proper mount point detection */
+	if ((s->oid.port != mnt.port) || (s->oid.id != mnt.id))
```

Comment:
seems like something we should do with every access to the file? (are current changes enough to eg. remove file from mounted partition?

If it's always needed, it should be taken care of inside lookup IMHO

Reply:
@nalajcie Lookup is a universal function - it returns both file/dir and device/fs mounted on this file/dir. It is responsibility of a lookup user to one oid or another (do we intend to access the file/dir or a dev/fs mounted there?). If nothing is mounted then both oids are equal. What would you like to change about lookup behaviour?

Reply:
I assume the problem is that if a file is on a mounted filesystem in `/`,  the `dir` points to the directory oid in the outer system instead of the `/` in the mounted partiton (so is actually invalid, as this file does not exist in the `dir` returned by the lookup?)